pkg/pod/entrypoint_lookup_impl.go

```diff
@@ -70,32 +58,71 @@ func (e *entrypointCache) Get(ctx context.Context, ref name.Reference, namespace
 		return nil, fmt.Errorf("error creating k8schain: %v", err)
 	}
 	mkc := authn.NewMultiKeychain(kc)
```

Comment:
FYI a `NewMultiKeychain` of one thing is the same as just passing that one thing.